moto/core/models.py

```diff
@@ -252,12 +255,22 @@ def _url_matches(self, url, other, match_querystring=False):
 
 
 def _find_first_match(self, request):
+    match_failed_reasons = []
     for i, match in enumerate(self._matches):
-        if match.matches(request):
-            return match
-
-    return None
+        if LooseVersion(RESPONSES_VERSION) < LooseVersion('0.12.1'):
```

Comment:
For readability, would you mind rewriting this to only have the single if-statement? 
Whether it's a separate function or everything in the same function is the same to me.

Usage of 'pkg_resources' is fine IMO.

You might have to run 'black' as well afterwards, as Travis is currently complaining about it.